ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+|----------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `sender`       | `address` | The wallet making the intent                                                                                                                             |
+| `standard`     | `address` | The `IStandard` implementation to validate and parse this `UserIntent`                                                                                   |
+| `header`       | `bytes`   | The metadata of this `UserIntent`, used by `standard`, using `bytes` type to keep flexibility                                                            |
+| `instructions` | `bytes[]` | The detailed content of this `UserIntent`, used by `standard` to determine the `Operation`s need to be executing, using `bytes` type to keep flexibility |
+| `signatures`   | `bytes[]` | Validatable signatures, used by `standard`                                                                                                               |
+
+But using `bytes[]` on functional interface is heavy especially when multiple intents are compound together. Therefore,
+the "UserIntent" object is designed to be a pack-encoded `bytes calldata userIntent` contains the above information
+
+| Section                         | Value Type | Description                                                     |
+|---------------------------------|------------|-----------------------------------------------------------------|
+| userIntent[0:20]                | `address`  | sender                                                          |
+| userIntent[20:40]               | `address`  | standard                                                        |
+| userIntent[40:42]               | `uint16`   | Length of header                                                |
+| userIntent[42:44]               | `uint16`   | Length of instructions                                          |
+| userIntent[44:46]               | `uint16`   | Length of signatures                                            |
+| Next `headerLength` bytes       | `bytes`    | The actual header                                               |
+| Next `instructionsLength` bytes | `bytes`    | The actual instructions                                         |
+| Next `signatureLength` bytes    | `bytes`    | The actual signatures                                           |
+| Anything left                   | `bytes`    | Extra data, e.g. nested intent to be executed within the intent |
+
+The above schema is NOT REQUIRED for all implementations. Each `IAccount` and `IStandard` can fully determine their own way of handling the `bytes` data. In the above example, the fields can be used in this way
```

Comment:
You should make it clear _before_ introducing the table that the encoding is only an example and is not required by the standard. I'd even go a step further and put the optional content into a subsection.

Reply:
I've move the section before the table 🙏